addons/account/models/account_payment.py

```diff
@@ -183,13 +183,20 @@ def _seek_for_lines(self):
         counterpart_lines = self.env['account.move.line']
         writeoff_lines = self.env['account.move.line']
 
-        for line in self.move_id.line_ids:
-            if line.account_id in self._get_valid_liquidity_accounts():
-                liquidity_lines += line
-            elif line.account_id.internal_type in ('receivable', 'payable') or line.account_id == line.company_id.transfer_account_id:
-                counterpart_lines += line
-            else:
-                writeoff_lines += line
+        if len(self.move_id.line_ids) == 2:
+            for line in self.move_id.line_ids:
+                if line.account_id.internal_type in ('receivable', 'payable') or line.account_id == self.company_id.transfer_account_id:
+                    counterpart_lines += line
+                else:
+                    liquidity_lines += line
+        else:
+            for line in self.move_id.line_ids:
+                if line.account_id in self._get_valid_liquidity_accounts():
+                    liquidity_lines += line
+                if line.account_id.internal_type in ('receivable', 'payable') or line.account_id == self.company_id.transfer_account_id:
+                    counterpart_lines += line
+                else:
+                    writeoff_lines += line
```

Comment:
@smetl something like that?

Reply:
@guva-odoo 
```py
nb_lines = len(self.move_id.line_ids)
for i, line in enumerate(self.move_id.line_ids):
    is_last = i == nb_lines - 1
    if is_last and nb_lines == 2:
        if liquidity_lines:
            counterpart_lines += line
        elif counterpart_lines:
            liquidity_lines += line
    elif line.account_id in self._get_valid_liquidity_accounts():
        liquidity_lines += line
    elif line.account_id.internal_type in ('receivable', 'payable') or line.account_id == self.company_id.transfer_account_id:
        counterpart_lines += line
    else:
        writeoff_lines += line
```

Reply:
@smetl it works fine except I had to sort `self.move_id.line_ids` to ensure that we have the receivable line first, if we don't do that the first line is assigned to `writeoff_lines` and we end up with no `liquidity_lines` and no `counterpart_lines`